### PR TITLE
Easy Coding Standard integration, cleanup

### PR DESCRIPTION
**Pull request by bobdenotter**

### CODE REVIEW

File: src/Json.php [hunk 1]
```diff
@@ -88,7 +85,8 @@ public static function parse($json, $options = 0, $depth = 512)
 
         $data = @json_decode($json, true, $depth, $options);
 
-        if ($data === null && ($json === '' || ($code = json_last_error()) !== JSON_ERROR_NONE)) {
+        if ($data === null && ($json === '' || (json_last_error()) !== JSON_ERROR_NONE)) {
```
Comment (JarJak) [inline src/Json.php:88]: why not use \GuzzleHttp\json_decode() or any other throwing json decorator?

File: src/Serialization.php [hunk 2]
```diff
@@ -15,18 +17,14 @@ class Serialization
...
-     * @return string
      */
-    public static function dump($value)
+    public static function dump($value): string
     {
         try {
             return serialize($value);
         } catch (\Error $e) {
-        } catch (\Exception $e) {
+        } catch (\Throwable $e) {
```
Comment (JarJak) [inline src/Serialization.php:26]: Throwable catches everything, so previous catch is useless

File: src/Serialization.php [hunk 3]
```diff
@@ -54,7 +50,7 @@ public static function parse($value, $options = [])
         } catch (ParseException $e) {
             throw $e;
         } catch (\Error $e) {
-        } catch (\Exception $e) {
+        } catch (\Throwable $e) {
```
Comment (JarJak) [inline src/Serialization.php:51]: same here

File: src/Str.php [hunk 4]
```diff
@@ -1,73 +1,49 @@
...
+    public static function replaceFirst(string $subject, string $search, string $replace, bool $caseSensitive = true): string
     {
-        $pos = $caseSensitive ? strpos($subject, $search) : stripos($subject, $search);
+        $pos = $caseSensitive ? mb_strpos($subject, $search) : mb_stripos($subject, $search);
         if ($pos === false) {
             return $subject;
         }
 
-        return substr_replace($subject, $replace, $pos, \strlen($search));
+        return substr_replace($subject, $replace, $pos, \mb_strlen($search));
```
Comment (JarJak) [inline src/Str.php]: keep using \ before global functions or not, but please stick with one approach

**Review by JarJak — CHANGES_REQUESTED** [review]

File: src/Json.php (hunk 1 above)
Comment (bobdenotter) [reply src/Json.php:88]: Not sure. This wasn't written by me. Maybe we should eventually, but for now I just wanted to get ECS in, and fix that specific issue

File: src/Str.php (hunk 4 above)
Comment (bobdenotter) [reply src/Str.php]: Fair enough.. Do you know if there's a quick-fix for this using ECS?

Comment (JarJak) [reply src/Str.php]: Unfortunately not. Maybe there is one, but I don't know the name.

Comment (bobdenotter) [reply src/Str.php]: I replaced all instances on one file, and i didn't like how it looked:

```php
-\mb_strlen($search) // Eeeewww
```

Reading up on it, I see Symfony decided against doing it, so i think we should follow suit. 

I'll remove the few we have, for consistency.

Comment (bobdenotter) [reply src/Str.php]: Link for reference: https://github.com/symfony/symfony/issues/21020

File: src/Serialization.php (hunk 3 above)
Comment (bobdenotter) [reply src/Serialization.php:51]: Fixed

File: .travis.yml [hunk 5]
```diff
@@ -24,12 +24,8 @@ install:
   - composer install --no-interaction $COMPOSER_FLAGS
 
 script:
-  # PHPUnit
-  - vendor/bin/phpunit --coverage-clover=coverage.xml
-  # PHP CodeSniffer
-  - vendor/bin/phpcs -p --report=summary
-  # PHP CS Fixer
-  - PHP_CS_FIXER_IGNORE_ENV=1 vendor/bin/php-cs-fixer fix --cache-file cs_cache/php_cs_fixer --dry-run
+  - ./vendor/bin/phpunit --coverage-clover=coverage.xml
```
Comment (JarJak) [inline .travis.yml]: >Error:         No code coverage driver is available

It needs XDebug installed to work :) Otherwise remove coverage report

**Review by JarJak — APPROVED** [review]

File: .travis.yml (hunk 5 above)
Comment (bobdenotter) [reply .travis.yml]: I was wondering what that was for.. ;-)

Updated!

**Pull request merged**